tests/pytests/test_scf_options.py

```diff
@@ -38,7 +43,99 @@ def test_guess_mix_for_broken_symmetry(inp):
     psi4.set_options({"guess_mix": True})
     thisBSSCF = psi4.energy("scf")
 
-    assert compare_values(refENuc, h2.nuclear_repulsion_energy(), 10, "Nuclear repulsion energy")
+    assert compare_values(
+        refENuc, h2.nuclear_repulsion_energy(), 10, "Nuclear repulsion energy"
+    )
     assert compare_values(refSCF, thisSCF, 10, "Reference energy")
     assert compare_values(refBSSCF, thisBSSCF, 10, "Reference broken-symmetry energy")
 
+
+@pytest.mark.parametrize("ref", ["rhf", "uhf", "rohf"])
+@pytest.mark.parametrize(
+    "inp",
+    [
+        pytest.param({"options": {"guess": "core"}}, id="core"),
+        pytest.param({"options": {"guess": "gwh"}}, id="gwh"),
+        pytest.param({"options": {"guess": "huckel"}}, id="huckel"),
+        pytest.param({"options": {"guess": "modhuckel"}}, id="modhuckel"),
+        pytest.param({"options": {"guess": "sad"}}, id="sad"),
+        pytest.param({"options": {"guess": "sadno"}}, id="sadno"),
+        pytest.param({"options": {"guess": "sap"}}, id="sap"),
+        # pytest.param({'options': {"guess": "sapgau"}}, id="sapgau"),
+    ],
+)
+def test_scf_guess(inp, ref):
+    vals = {
+        "rhf": {
+            "core": -85.33505416384983,
+            "gwh": -92.48691177199250,
+            "sad": -99.63941801281894,
+            "sadno": -99.77439003262768,
+            "huckel": -99.82915309755559,
+            "modhuckel": -99.80751245771735,
+            "sap": -99.97316521623380,
+        },
+        "uhf": {
+            "core": -88.99202896495355,
+            "gwh": -94.35585144644561,
+            "sad": -99.63941801281894,
+            "sadno": -99.11882081641045,
+            "huckel": -99.20374047108815,
+            "modhuckel": -99.17689049002892,
+            "sap": -99.50144270485751,
+        },
+        "rohf": {
+            "core": -88.99202896495353,
+            "gwh": -94.35585144644557,
+            "sad": -99.63941801281894,
+            "sadno": -99.11882081641042,
+            "huckel": -99.20374047108817,
+            "modhuckel": -99.17689049002892,
+            "sap": -99.50144270485754,
+        },
+    }
+    ref_final = {
+        # "refnuc": 5.282967161430950,
+        "rhf": -100.0584459442408587,
+        "uhf": -99.5312257221445549,
+        "rohf": -99.5261713512123976,
+        "cuhf": -99.5261713512123976,
+    }
+    options = {
+            "reference": ref,
+            "basis": "cc-pvtz",
+            "scf_type": "pk",
+            "df_scf_guess": False,
+            "guess": inp["options"]["guess"],
+        }
+    if ref == "rhf":
+        hf = psi4.geometry(
+            """
+    0 1
```

Comment:
I know the spacing here was your autoformatter's fault, but probably better to have this indented properly.

Reply:
Mol formatting is pretty resilient.